packages/@tailwindcss-upgrade/src/template/prepare-config.ts

```diff
@@ -94,15 +94,36 @@ const DEFAULT_CONFIG_FILES = [
   './tailwind.config.cts',
   './tailwind.config.mts',
 ]
-export async function detectConfigPath(base: string) {
-  for (let file of DEFAULT_CONFIG_FILES) {
-    let fullPath = path.resolve(base, file)
-    try {
-      await fs.access(fullPath)
-      return file
-    } catch {}
+export async function detectConfigPath(start: string, end: string = start) {
+  for (let base of parentPaths(start, end)) {
+    for (let file of DEFAULT_CONFIG_FILES) {
+      let fullPath = path.resolve(base, file)
+      try {
+        await fs.access(fullPath)
+        return fullPath
+      } catch {}
+    }
   }
+
   throw new Error(
-    'No configuration file found. Please provide a path to the Tailwind CSS v3 config file via the `--config` option.',
+    `No configuration file found for ${highlight(relative(start))}. Please provide a path to the Tailwind CSS v3 config file via the ${highlight('--config')} option.`,
   )
 }
+
+// Yields all parent paths from `from` to `to` (inclusive on both ends)
+function* parentPaths(from: string, to: string = from) {
+  let fromAbsolute = path.resolve(from)
```

Comment:
Should this have a check to ensure that `from.startsWith(to)`? otherwise this might never terminate